### ISSUE TITLE
In Correct content order for clarity signpost component read by screen reader.

### ISSUE THREAD

**Issue by rahultokase** [vmware/clarity#4986]
## Describe the accessibility issue

Clarity signpost content element create the following two element which are not in order hence the correct order is not been read by the screen reader.

`<clr-signpost-content _ngcontent-tli-c4="" class="top-middle signpost-content" ng-reflect-position="top-middle" id="clr-id-5" style="position: absolute; top: 0px; bottom: auto; left: 0px; right: auto; transform: translateX(-25px) translateY(36px);"><div class="signpost-wrap"><div class="popover-pointer"></div><div class="signpost-content-body"> Lorem ipsum... </div><div class="signpost-content-header"><button class="signpost-action close" type="button" aria-label="Signpost Close" aria-controls="clr-id-5"><clr-icon shape="close" title="Close"></clr-icon></button></div></div></clr-signpost-content>`

the clarity signpost element is rendering the .signpost-content-body element first and then .signpost-action close element. Hence the content is not read in the order visible to the human eyes. The correct order should be close button-> clarity signpost body by press Down arrow key.

_If this is a VMware product, do not fill this out! Instead consult with the accessibility team about the issue so we can consolidate all VMware related issues and track them separately._

_Provide clear description of the accessibility issue and the impact to the user. Is this only in certain use cases? Is it confusing or broken?_
Vision-impaired users may get confused when signpost content is been shown to user.

## WCAG Guideline

_What WCAG guideline is being violated here and how? You should reference a guideline from https://www.w3.org/TR/WCAG21/_
WCAG 2.1 -> 1.3.1 Info and Relationships (Level A):  
Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text.  

## Recommended Solution:  
Move the <div class="signpost-content-header"> up, to make it as the first element in <div class="signpost-content-body">

## How to reproduce

_Please provide a link to a reproduction scenario using one of the [Clarity Stackblitz templates](https://stackblitz.com/@clr-team). Reports without a clear reproduction may not be prioritized until one is provided._
https://stackblitz.com/edit/angular-8-app-clarity-signpost-issue?file=src%2Fapp%2Fapp.component.html

1.  Go to '...'
2.  Click on '....'
3.  Scroll down to '....'
4.  See error

## Expected behavior

_A clear and concise description of what you expected to happen._
\*Recommended Solution:  
Move the <div class="signpost-content-header"> up, to make it as the first element in <div class="signpost-content-body">

## Versions

**App**

- Angular: [e.g. 6] 8
- Node: [e.g. 8.10.0] 10.16.0
- Clarity: [e.g. 0.12.5] 2.4.0

**Device:**

- Type: [e.g. MacBook]
![Signpost content ordering](https://user-images.githubusercontent.com/5329586/91731615-aace8500-ebc4-11ea-88ee-43cac97d20bb.png)
Win10 64bit
- Browser [e.g. chrome, safari] Chrome 
- Version [e.g. 22] 84.0.4147.89
- Screen Reader: [e.g. VoiceOver, NVDA] NVDA 2020.1

For screen readers, we primarily support the following combinations, and other combinations will be considered on a case by case basis.

- NVDA + Firefox (Windows)
- VoiceOver + Safari (Apple)

## Additional notes

_Add any other notes about the problem here._

**Comment by Jinnie** [comment]
@chlane Can we have your comment on this issue?
The W3C examples of popups with close button, which I am able to find, feature order similar to the one in our signpost.
The only difference I find is that our Close button is vertically located in the padding above the body content, while in the examples it stays adjacent to the text.
Like here:
https://www.w3.org/WAI/WCAG21/working-examples/aria-label-invisible-label-box/

**Comment by chlane** [comment]
I agree that this issue is valid. @Jinnie Thanks for finding this example - https://www.w3.org/WAI/WCAG21/working-examples/aria-label-invisible-label-box/. In this example, from a left to right reading order perspective, the content "This is a div box." visually appears to come before the close button. This matches how the elements are arranged in the DOM. But in the signpost, the close button visually appears to be the first element because it is visually positioned above the signpost content.

**Comment by amellnik** [comment]
Would it be sufficient to use `aria-owns` to reoder the content in the a11y without changing the DOM order at all?

**Comment by chlane** [comment]
@amellnik You could try it but support is inconsistent. When you try it, test to ensure that it works in macOS/VoiceOver/Safari and Windows/NVDA/Firefox. I personally recommend against until it supported by all assistive technologies.

See 
https://labs.levelaccess.com/index.php/ARIA_List_using_aria-owns
https://tink.uk/using-the-aria-owns-attribute/

**Comment by amellnik** [comment]
Looking at the [browser and screen reader support list] I think we should be OK.  I have easy access to a Windows system for NVDA testing but may need to reach out for some help on the VoiceOver front (although I understand that recent versions work fine with `aria-owns`).

**Comment by amellnik** [comment]
Update: In this case it was trivial to just reorder the divs so no `aria-owns` was necessary.

**Comment by bbogdanov** [comment]
The fix will be available in `2.1.8` `3.1.11` and `4.0.5`
